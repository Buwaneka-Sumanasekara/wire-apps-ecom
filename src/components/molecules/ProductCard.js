import React from "react";
import { Image, Text, TouchableOpacity, View } from "react-native";
import AddToCartButton from "../atoms/AddToCartButton";
import withProductContext from "../../templates/withProductContext";


const ProductCard = ({ item, onPressItem }) => {

    return <TouchableOpacity onPress={() => onPressItem(item)} className={"w-1/2 px-1 pb-2"}>
        <View className={"flex-1  bg-white rounded-md justify-content-center p-3"}>
            <Image source={{ uri: item.mainImage }} className={"w-3/4 aspect-square self-center"} />
            <Text className={"text-base text-black font-semibold"}>{`${item.price.amount} ${item.price.currency}`}</Text>
            <View className={"flex flex-row"}>
                <Text ellipsizeMode={"tail"} numberOfLines={2} className={"text-sm text-slate-500 w-4/5"}>{item.name}</Text>
                <AddToCartButton productId={item.id} isMini={true} />
            </View>
        </View>
    </TouchableOpacity>;

}

export default withProductContext(ProductCard);