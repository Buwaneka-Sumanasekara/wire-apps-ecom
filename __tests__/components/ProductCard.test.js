import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../../src/components/molecules/ProductCard';
import productJson from '../data/product.json';

describe('<ProductCard/>', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<ProductCard item={productJson} />).toJSON()
        expect(tree).toMatchSnapshot()
    });
});

