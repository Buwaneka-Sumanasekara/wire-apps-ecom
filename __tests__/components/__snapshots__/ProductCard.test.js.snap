// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard/> rendered sucessfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 4,
      "paddingRight": 4,
      "width": "50%",
    }
  }
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "backgroundColor": "#fff",
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        {
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Image
      source={
        {
          "uri": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/7e386191b2ee40b290886a05d3e10e24_nike-air-relentless-a.jpg",
        }
      }
      style={
        [
          {
            "width": "75%",
          },
          {
            "aspectRatio": 1,
          },
          {
            "alignSelf": "center",
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "color": "#000",
          },
          {
            "fontWeight": "600",
          },
        ]
      }
    >
      45 GBP
    </Text>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "color": "#64748b",
            },
            {
              "width": "80%",
            },
          ]
        }
      >
        Nike Air Relentless 4 Mens Running Shoes
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
